recipes/cuda-nvcc-impl/meta.yaml

```diff
@@ -0,0 +1,167 @@
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: Architecture independent part of CUDA NVCC compiler.
+      description: |
+        Compiler for CUDA applications.
+      doc_url: https://docs.nvidia.com/cuda/index.html
+
+  - name: cuda-nvcc-dev_{{ target_platform }}
+    build:
+      noarch: generic
+    run_exports:
+      strong:
+        - cuda-version >={{ cuda_version }},<{{ cuda_version.split(".")[0]|int + 1 }}
+    files:
+      - targets/{{ target_name }}/include  # [linux]
+      - targets/{{ target_name }}/lib      # [linux]
+      - Library\bin      # [win]
+      - Library\nvvm     # [win]
+      - Library\include  # [win]
+      - Library\lib      # [win]
+    requirements:
+      build:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+      host:
+        - cuda-version {{ cuda_version }}
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+      run_constrained:
+        - gcc_impl_{{ target_platform }} >=6,<13  # [linux]
+        - arm-variant * {{ arm_variant_type }}    # [aarch64]
+    test:
+      commands:
+        - test -f $PREFIX/targets/{{ target_name }}/lib/libnvptxcompiler_static.a     # [linux]
+        - test -f $PREFIX/targets/{{ target_name }}/include/nvPTXCompiler.h           # [linux]
+        - if not exist %LIBRARY_LIB%\nvptxcompiler_static.lib exit 1  # [win]
+        - if not exist %LIBRARY_INC%\nvPTXCompiler.h exit 1           # [win]
+    about:
+      home: https://developer.nvidia.com/cuda-toolkit
+      license_file: LICENSE
+      license: LicenseRef-NVIDIA-End-User-License-Agreement
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: Target architecture dependent parts of CUDA NVCC compiler.
+      description: |
+        Compiler for CUDA applications.
+      doc_url: https://docs.nvidia.com/cuda/index.html
+
+  - name: cuda-nvcc-impl
+    run_exports:
+      strong:
+        - cuda-version >={{ cuda_version }},<{{ cuda_version.split(".")[0]|int + 1 }}
+    files:
+      - include/crt                    # [linux]
+      - include/fatbinary_section.h    # [linux]
+      - include/nvPTXCompiler.h        # [linux]
+      - lib/libnvptxcompiler_static.a  # [linux]
+      - nvvm/include                   # [linux]
+      - nvvm/lib64                     # [linux]
+    requirements:
+      build:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - sysroot_{{ target_platform }} 2.17    # [linux]
+      host:
+        - cuda-version {{ cuda_version }}
+        - cuda-cudart-dev
+      run:
+        - {{ pin_subpackage("cuda-nvcc-tools", exact=True) }}
+        - {{ pin_subpackage("cuda-nvcc-dev_" + target_platform, exact=True) }}
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+        - cuda-cudart-dev
+      run_constrained:
+        - gcc_impl_{{ target_platform }} >=6,<13  # [linux]
+        - arm-variant * {{ arm_variant_type }}    # [aarch64]
+    test:
```

Comment:
Shall we move the cmake tests from cuda-nvcc-feedstock here?

Reply:
They rely on the activation scripts, so I was thinking they are better left in one of those metapackages.

Reply:
We can just install the `gxx` package and it should just work.

Reply:
`nvcc` isn't finding the host compiler.

In my test environment I see these installed:
```
    gxx_impl_linux-64:           12.2.0-hcc96c02_19      conda-forge
    gxx_linux-64:                12.2.0-hb41e900_13      conda-forge
```
and the error:
```
CMake Error at /home/conda/staged-recipes/build_artifacts/cuda-nvcc-impl-split_1683916210951/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/share/cmake-3.26/Modules/CMakeDetermineCompilerId.cmake:751 (message):
  Compiling the CUDA compiler identification source file
  "CMakeCUDACompilerId.cu" failed.

  The output was:
  1
  gcc: No such file or directory
  nvcc fatal : Failed to preprocess host compiler properties.
```

Reply:
Did you have the `gxx` and `gcc` packages installed?

Reply:
Yes `gcc` as well:
```
    gcc_impl_linux-64:           12.2.0-hcc96c02_19      conda-forge
    gcc_linux-64:                12.2.0-h4798a0e_13      conda-forge
```

Reply:
I mean `gxx` not `gxx_linux-64`.

Reply:
Thanks, that got it working for me now. Although I did have to revert this [nvcc.profile change](https://github.com/conda-forge/staged-recipes/pull/22802/commits/bdbc5ece9acde348806e0872b9f31ebe64db523c#diff-3cee1670430a2dad82a3fd348804d371fb07cb4be2f198fc61dc9fa76a292e82) to get it to find `nvvm` correctly